### PR TITLE
sla_tests: use SL REST API instead of binary protocol on/off workaround

### PR DESCRIPTION
**Pull request by Jadw1**
Current workaround with restarting binary protocol to update connections' scheduling group couldn't be done on all nodes because of running nemeses, which resulted in problems like https://github.com/scylladb/scylla-enterprise/issues/2572

This patch changed the method to use recently created REST API endpoints https://github.com/scylladb/scylla-enterprise/pull/3242 (`/service_levels/switch_tenants` and `/service_levels/count_connections`) to trigger scheduling groups update and to validate them.

### **Note** 
**This PR to run correctly requires https://github.com/scylladb/scylla-enterprise/pull/3242 to be in scylla and none of the 2023.x 2024.x have that PR.**

### CODE REVIEW

File: sdcm/sla/sla_tests.py
Comment (temichus) [inline sdcm/sla/sla_tests.py:127]: restart_scylla  after this PR will be unused. so it needs to be deleted from "attach_sl_and_validate_scheduler_runtime" method and it usage

File: sdcm/sla/sla_tests.py
Comment (temichus) [inline sdcm/sla/sla_tests.py:133]: what we are waiting for here? Could you please leave a comment?  Can we make some smart wait? for example by periodically checking service_levels/count_connections or it is not the case here?

**Comment by fruch** [comment]
@Jadw1 
see the linting errors in:
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-6950/2/execution/node/23/log

**Comment by fruch** [comment]
@Jadw1 

can you share a run demonstrating this works ?, also can you clarify in which release its gonna be available ?

File: sdcm/sla/sla_tests.py
Comment (juliayakovlev) [inline sdcm/sla/sla_tests.py:123]: Do you plan to handle this issue (https://github.com/scylladb/scylla-enterprise/issues/2572)? If yes - we still need the comment that it is workaround. So when the issue will be fixed - we will remove it from the code

**Review by juliayakovlev — COMMENTED** [review]
Generally it looks ok. Small comment

**Comment by juliayakovlev** [comment]
I ran the test with this commit (uses API) and received the result worse then when binary disable/enable is used as workaround.

Test ran with **SL REST API**:
https://argus.scylladb.com/test/d10575b9-86ef-485a-8bd7-089f423436d3/runs?additionalRuns[]=5426c441-6d6f-4cfb-8fe9-f10a91585e0f

test ran with **binary disable/enable**:
https://argus.scylladb.com/test/7bdbd517-fe3d-441b-aa66-a581404f305a/runs?additionalRuns[]=44214055-422a-4038-bd2e-bb9efcdc789d

**Comment by juliayakovlev** [comment]
The issue was reproduced with using this commit: https://github.com/scylladb/scylla-enterprise/issues/2572#issuecomment-1873796736

**Comment by Jadw1** [comment]
> @Jadw1
> 
> can you share a run demonstrating this works ?, also can you clarify in which release its gonna be available ?

Run with the PR: https://argus.scylladb.com/test/4b2b31e3-1d72-4af3-a281-cb436db14cc2/runs?additionalRuns[]=7a69fcb1-3f77-4982-8309-a95d370437fc

Run without the PR: https://argus.scylladb.com/test/5e62ce40-3da5-4002-9260-dbc5e545e1d4/runs?additionalRuns[]=af3a15a5-7630-41f4-a501-4f2dd27a0d78

Both those run was on the same scylla version (quite latest enterprise). Two of the nemesis failed on run with this PR but this might be due to https://github.com/scylladb/scylla-enterprise/issues/3737#issuecomment-1911024842. But all of the connections are working under correct scheduling group (the information is in sct runner's logs) and this is what this PR is fixing.
Without the PR 4 nemsis failed.

Note that for this PR to run correctly, https://github.com/scylladb/scylla-enterprise/pull/3242 has to be in scylla and **none of the 2023.x 2024.x have that PR**. That's why I've run the test against current enterprise version.
Maybe we should backport service levels REST API PR before merging this PR? Cc @eliransin @mykaul

**Comment by Jadw1** [comment]
precommit failed but I don't see any clear error in jenkins logs. And precommit was successful locally on my machine

File: sdcm/sla/sla_tests.py
Comment (Jadw1) [reply sdcm/sla/sla_tests.py:127]: removed

File: sdcm/sla/sla_tests.py
Comment (Jadw1) [reply sdcm/sla/sla_tests.py:124]: added a comment

**Comment by vponomaryov** [comment]
> precommit failed but I don't see any clear error in jenkins logs. And precommit was successful locally on my machine

Jenkins logs are following:
```
13:43:04  ⧗   input: sla_tests: use SL REST API instead of binary protocol on/off workaround
13:43:04  ✖   type must be one of [ci, docs, feature, fix, improvement, perf, refactor, revert, style, test, unit-test] [type-enum]
13:43:04  ✖   scope may not be empty [scope-empty]
```

So, it means that the commit topic violates expected structure.
It must be like following:
```
fix(sla_tests): use SL REST API instead of binary protocol on/off workaround
```

**Comment by Jadw1** [comment]
@temichus @juliayakovlev @fruch Can we merge this?

**Comment by juliayakovlev** [comment]
@Jadw1 which versions https://github.com/scylladb/scylla-enterprise/pull/3242 commit is / will be backported to?

**Comment by Jadw1** [comment]
None of the versions have it. I guess we should backport it to 2024.1, idk how about 2023.
@mykaul @kostja Can you confirm/answer?

**Comment by juliayakovlev** [comment]
> None of the versions have it. I guess we should backport it to 2024.1, idk how about 2023. @mykaul @kostja Can you confirm/answer?

@Jadw1 
If so - we can not merge this commit. It will fail the test. Or I miss something?

**Comment by kostja** [comment]
@juliayakovlev I think we should not backport it unless there is an impact on the customer. Do you have a reason to backport it? Could you move this sct to run with the master instead?

**Comment by mykaul** [comment]
> @juliayakovlev I think we should not backport it unless there is an impact on the customer. Do you have a reason to backport it? Could you move this sct to run with the master instead?

We are not going to backport it - it should indeed run on master.

**Comment by fruch** [comment]
> > @juliayakovlev I think we should not backport it unless there is an impact on the customer. Do you have a reason to backport it? Could you move this sct to run with the master instead?
> 
> We are not going to backport it - it should indeed run on master.

this test isn't running regularly on master, it running as part of a release.
the question was mostly to understand where we should backport this one.

**Comment by fruch** [comment]
> > > @juliayakovlev I think we should not backport it unless there is an impact on the customer. Do you have a reason to backport it? Could you move this sct to run with the master instead?
> > 
> > 
> > We are not going to backport it - it should indeed run on master.
> 
> this test isn't running regularly on master, it running as part of a release. the question was mostly to understand where we should backport this one.

FYI, whom would run SLA with master SCT, might run into it.
whom tests SLA, should be aware of it, otherwise he's in for a supprise.

**Review by fruch — APPROVED** [review]
LGTM

**Comment by juliayakovlev** [comment]
If th

> @juliayakovlev I think we should not backport it unless there is an impact on the customer. Do you have a reason to backport it? Could you move this sct to run with the master instead?

@kostja @fruch 
if this rest api won't be part of version, how we can run the test and use it?

**Comment by kostja** [comment]
It is part of master, so will be part of a future release. The question was do you need it for an older release? I'd presume no, unless there is a good reason to run this test with this old release (please share it).

**Pull request merged**
